File: 3754. Concatenate Non-Zero Digits and Multiply by Sum I.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
       s=0
       x=0
       power=0
       while(n>0):
        temp=n%10
        if(temp!=0):
            x=(temp*(10**power))+x
            power+=1
        
        s+=temp
        n//=10
    
       return s*x

File: 3754. Concatenate Non-Zero Digits and Multiply by Sum I.cpp
class Solution {
public:
    long long sumAndMultiply(int n) {
        long long sum=0;
        long long x=0;
        long long counter=0;
        while(n>0)
        {
            int temp=n%10;
            if(temp!=0)
            {
                x=temp*pow(10,counter)+x;
                counter++;
            }
            sum+=temp;
            n/=10;
        }
        int temp=x;
        return sum*x;
    }
};